elixir/apps/web/lib/web/live/gateways_live/index.ex

```diff
@@ -1,6 +1,25 @@
 defmodule Web.GatewaysLive.Index do
   use Web, :live_view
 
+  alias Domain.Gateways
+  alias Domain.Resources
+
+  def mount(_params, _session, socket) do
+    subject = socket.assigns.subject
+    {_, gateways} = Gateways.list_gateways(subject, preload: :group)
+
+    {_, resources} =
+      Enum.map_reduce(gateways, %{}, fn g, acc ->
+        {_, count} = Resources.count_resources_for_gateway(g, subject)
+        {count, Map.put(acc, g.id, count)}
+      end)
+
+    grouped_gateways = Enum.group_by(gateways, fn g -> g.group end)
+
+    {:ok,
+     assign(socket, gateways: gateways, resources: resources, grouped_gateways: grouped_gateways)}
```

Comment:
Let's have a rule of thumb: if there is more than one assign, then function should be moved out of tuple to make code more readable:
```suggestion
    socket = 
      assign(socket, 
        gateways: gateways, 
        resources: resources, 
        grouped_gateways: grouped_gateways
      )
    
    {:ok, socket}
```

Reply:
👍 makes sense.  I just realized that I don't think I need the `gateways: gateways` assignment, so I'll push an updated commit and reformat things like you have them here.

Reply:
I wish we can report that is assignment is not used in heex, for LiveView reducing state is pretty important. (If you are up for a challenge I can help and you can contribute this to Phoenix :).

Reply:
That would be awesome!

Reply:
This has been updated to remove the gateways assignment and the `assign` call has been moved out of the tuple.